source/DSCResources/DSC_CMFileReplication/DSC_CMFileReplication.psm1

```diff
@@ -0,0 +1,800 @@
+            $valuesToCheck = @('Unlimited')
+        }
+
+        if ($state.Ensure -eq 'Absent')
+        {
+            Write-Verbose -Message $script:localizedData.FileRepAbsent
+            $return = $false
+        }
+        else
+        {
+            $valuesToCheck += @('FileReplicationAccountName')
+            $testParams = @{
+                CurrentValues = $state
+                DesiredValues = $PSBoundParameters
+                ValuesToCheck = $valuesToCheck
+            }
+
+            $return = Test-DscParameterState @testParams -TurnOffTypeChecking -Verbose
+
+            if ($checkSchedule -eq $true)
+            {
+                foreach ($limSched in $RateLimitingSchedule)
+                {
+                    $overlapCheck = $RateLimitingSchedule.Where({$limSched.LimitedBeginHour -gt $_.LimitedBeginHour -and
+                                                                $limSched.LimitedBeginHour -lt $_.LimitedEndHour})
+                    if ($overlapCheck)
+                    {
+                        Write-Warning ($script:localizedData.OverlappingRate -f $limSched.LimitedBeginHour, $limSched.LimitedEndHour)
+                    }
+                    else
+                    {
+                        $compliance = $false
+
+                        foreach ($thing in $state.RateLimitingSchedule)
+                        {
+                            if (($limSched.LimitedBeginHour -ge $thing.LimitedBeginHour -and ($limSched.LimitedEndHour -le $thing.LimitedEndHour -or $thing.LimitedEndHour -eq 0)) -and
+                                $limSched.LimitAvailableBandwidthPercent -eq $thing.LimitAvailableBandwidthPercent)
+                            {
+                                $compliance = $true
+                                break
+                            }
+                        }
+
+                        if ($compliance -eq $true)
+                        {
+                            Write-Verbose -Message ($script:localizedData.LimitSchedMatch -f $limSched.LimitedBeginHour, $limSched.LimitedEndHour,
+                                                    $limSched.LimitAvailableBandwidthPercent)
+                        }
+                        else
+                        {
+                            Write-Verbose -Message ($script:localizedData.LimitSchedNonMatch -f $limSched.LimitedBeginHour,$limSched.LimitedEndHour,
+                                                    $limSched.LimitAvailableBandwidthPercent)
+                            $return = $false
+                        }
+                    }
+                }
+            }
+
+            if ($NetworkLoadSchedule)
+            {
+                foreach ($limNetSched in $NetworkLoadSchedule)
+                {
+                    $loadOverlap = $NetworkLoadSchedule.Where({$_.Day -eq $limNetSched.Day -and $limNetSched.BeginHour -gt
+                        $_.BeginHour -and $limNetSched.BeginHour -lt $_.EndHour})
```

Comment:
$limNetSched.BeginHour -ge

Reply:
issues with condition evaluation need troubleshot. Day does not seem to be evaluated